### PR TITLE
Replace radio buttons for 'Role' field

### PR DESCRIPTION
**Pull request by patrick-du**
[Notion ticket](https://www.notion.so/uwblueprintexecs/replace-radio-buttons-817b05db5c324c7194e911b6162c10dd)

## Changes
- Replaced radio buttons for the 'Role' field within the `Add New Team Member` and `Edit Team Member` flows

## Screenshots
Before
image
image
After
image
image

## Testing
- Ensure that switching roles still works as expected in `Add a new team member` page and the `Edit team member` page

before opening PR
- [x] check notion ticket
- [x] run linter
- [x] go through file diff

filling out PR
- [x] descriptive title
- [x] update `notion ticket` link
- [ ] update `figma link`
- [x] fill out `Changes`
  - [ ] \(optional) add inline comments in diff
- [x] fill out `Testing`
- [x] \(optional) add `Screenshots`
- [x] assign yourself to the PR

after opening PR
- [x] link PR to notion ticket
- [x] move ticket to `In PR`
- [x] make sure build passes
- [x] ping `@ps` in `#paramedics-dev`

### CODE REVIEW

**Review by petorsong — DISMISSED** [review]
👓 looking good

File: src/components/ResourceForm/AmbulanceFormPage.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ const AmbulanceFormPage = ({
       } = data.ambulances.find((ambulance) => ambulance.id === ambulanceId);
       setAmbulanceNumber(vehicleNumber.toString());
     }
-  }, [data, loading, mode]);
+  }, [data, ambulanceId, loading, mode]);
```
Comment (patrick-du) [inline src/components/ResourceForm/AmbulanceFormPage.tsx:82]: Change is to fix missing `useEffect` dependency (unrelated to PR)

File: src/components/ResourceForm/HospitalFormPage.tsx [hunk 2]
```diff
@@ -75,7 +75,7 @@ const HospitalFormPage = ({
       } = data.hospitals.find((hospital) => hospital.id === hospitalId);
       setHospitalName(name);
     }
-  }, [data, loading, mode]);
+  }, [data, hospitalId, loading, mode]);
```
Comment (patrick-du) [inline src/components/ResourceForm/HospitalFormPage.tsx:78]: Change is to fix missing useEffect dependency (unrelated to PR)

**Review by petorsong — APPROVED** [review]
thanks for getting those console errors!

**Review by carelynntsai — APPROVED** [review]
looks good 🍕!

**Comment by carelynntsai** [comment]
@petorsong Currently I can add a patient without selecting one of the roles, and then it defaults to the supervisor role. Just wondering if this is the desired behaviour.

**Comment by petorsong** [comment]
> @petorsong Currently I can add a patient without selecting one of the roles, and then it defaults to the supervisor role. Just wondering if this is the desired behaviour. 

Was that the behaviour we had before? If so I can check if there's a ticket for it (@patrick-du if it wasn't already implemented then you won't need to do it for this ticket)

**Comment by patrick-du** [comment]
> > @petorsong Currently I can add a patient without selecting one of the roles, and then it defaults to the supervisor role. Just wondering if this is the desired behaviour.
> 
> Was that the behaviour we had before? If so I can check if there's a ticket for it (@patrick-du if it wasn't already implemented then you won't need to do it for this ticket)

Yeah, this is the same behavior we had before.

**Pull request merged**